### PR TITLE
Unify (most) package metadata and tool configuration

### PR DESCRIPTION
**Pull request by daneah**
**This change is a:** (check at least one)
- [ ] Bugfix
- [ ] Feature addition
- [ ] Code style update
- [ ] Refactor
- [ ] Release activity
- [x] Infrastructure improvement

**Is this a breaking change?** (check one)
- [ ] Yes
- [x] No

**Is the:**
- [x] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [x] Test suite passing?
- [x] Code coverage maximal?
- [ ] Changelog up to date?

**What does this change address?**
Executable `setup.py` configuration has generally fallen out of favor, with declarative project configuration files like `setup.cfg` and `pyproject.toml` becoming more popular. Additionally, many tools have historically used tool-specific configuration files but are slowly converging on centralized configuration instead.

**How does this change work?**
The following files are subsumed by moving their configuration/information to `setup.cfg`:

* `.coveragerc`
* `MANIFEST.in`
* `apiron/VERSION` (thanks in part to `importlib_metadata` for the docs!)
* `pytest.ini`
* `requirements.txt`
* `tox.ini`

It also reduces `setup.py` to its minimal form.

**Additional context**
I think I would've preferred to cut over to `pyproject.toml` if we could, but [coverage does not yet support it](https://github.com/nedbat/coveragepy/pull/699). This also means the configuration for `black` still sits in `pyproject.toml`, because _it_ doesn't support `setup.cfg`. Fortunately, the content of `setup.cfg` should be relatively portable to `pyproject.toml` when the time comes.

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
--r requirements.txt
```
Comment (momo-sa) [inline dev-requirements.txt:1]: `requirements.txt` is no longer relevant to dev or are we taking a different approach?

Comment (daneah) [reply dev-requirements.txt:1]: `requirements.txt` has been removed in favor of `setup.cfg:options:install_requires`. `tox` packages and installs `apiron` in each virtual environment, so `requests` gets picked up as part of that process. You could optionally `pip install [-e] .` manually if you're doing interactive testing.

**Review by momo-sa — APPROVED** [review]

**Review by michael-iden — APPROVED** [review]

**Review by eslawski — APPROVED** [review]
Looks good. Nothing jumping out at me

**Review by NazimHAli — APPROVED** [review]
Looks good, was going to say we could add URLs to homepage or readthedoc, but this is meant to be minimal.

**Pull request merged**